flake.nix

```
@@ -0,0 +1,46 @@
+{
+  description = "A gdnative project with nix";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
+    rust-overlay.url = "github:oxalica/rust-overlay";
+    flake-utils.url = "github:numtide/flake-utils";
+  };
+
+  outputs = { self, nixpkgs, rust-overlay, flake-utils, ... }:
+    flake-utils.lib.eachDefaultSystem (system:
+      let
+        overlays = [
+          (import rust-overlay)
+
+        ];
+        pkgs = import nixpkgs {
+          inherit system overlays;
+        };
+      in
+      with pkgs;
+      {
+        devShells.default = mkShell.override { stdenv = pkgs.clangStdenv; } {
+          # Point bindgen to where the clang library would be
+          LIBCLANG_PATH = "${pkgs.libclang.lib}/lib";
+          # Make clang aware of a few headers (stdbool.h, wchar.h)
+          BINDGEN_EXTRA_CLANG_ARGS = with pkgs; ''
+            -isystem ${llvmPackages.libclang.lib}/lib/clang/${lib.getVersion clang}/include
+            -isystem ${llvmPackages.libclang.out}/lib/clang/${lib.getVersion clang}/include
+            -isystem ${glibc.dev}/include
+          '';
```

This is erroring in my Apple Silicon machine.
Don't know if darwin-x86 has the same issue but this can help:
```suggestion
          BINDGEN_EXTRA_CLANG_ARGS =
            if stdenv.isDarwin then with pkgs;
            ''
              -isystem ${stdenv.cc.cc}/include
              -isystem ${stdenv.cc.cc}/include/c++/v1
              -isystem ${stdenv.cc.cc}/include/clang-c
            ''
            else
              ''
                -isystem ${llvmPackages.libclang.lib}/lib/clang/${lib.getVersion clang}/include
                -isystem ${llvmPackages.libclang.out}/lib/clang/${lib.getVersion clang}/include
                -isystem ${glibc.dev}/include
              '';

```

---

Yeah `glibc` is not available for any `darwin` platform irrespective of intel/arm arch: https://search.nixos.org/packages?channel=unstable&show=glibc&from=0&size=50&sort=relevance&type=packages&query=glibc

I don't know if `godot3` supports `darwin`.


EDIT: ignore me it does not support.